src/main/java/sirius/web/security/UserInfo.java

```diff
@@ -286,42 +272,7 @@ public boolean hasPermissions(String... permissions) {
      * @return <tt>true</tt> if the user has the permission, <tt>false</tt> otherwise
      */
     public boolean hasPermission(String permission) {
-        if (Strings.isEmpty(permission)) {
-            return true;
-        }
-
-        if (DISABLED.equals(permission)) {
-            return false;
-        }
-
-        if (ENABLED.equals(permission)) {
-            return true;
-        }
-
-        for (String orClause : permission.split(",")) {
-            if (permissionsFullfilled(orClause)) {
-                return true;
-            }
-        }
-
-        return false;
-    }
-
-    protected boolean permissionsFullfilled(String permissions) {
-        for (String permission : permissions.split("\\+")) {
-            if (!permissionFullfilled(permission)) {
-                return false;
-            }
-        }
-        return true;
-    }
-
-    protected boolean permissionFullfilled(String permission) {
-        if (permission.startsWith("!")) {
-            return permissions == null || !permissions.contains(permission.substring(1));
-        } else {
-            return permissions != null && permissions.contains(permission);
-        }
+        return Permissions.hasPermission(permission, permissions == null ? s -> false : permission::contains);
```

Comment:
permissions:contains

Reply:
I wonder how this could pass the unit tests...

Reply:
Please change accordingly

Reply:
😱🤦‍♂️